Realm/Realm.Weaver/Analytics/AnalyticsUtils.cs

```diff
@@ -97,21 +106,26 @@ System.Environment.OSVersion.Platform switch
 
         public static (string Name, string Version) GetFrameworkAndVersion(ModuleDefinition module, Config config)
         {
-            if (config.TargetFramework.ContainsIgnoreCase("unity"))
+            if (config.UnityInfo != null)
             {
-                return config.TargetFramework.ContainsIgnoreCase("editor") ?
-                    (Framework.UnityEditor, config.TargetFrameworkVersion) :
-                    (Framework.Unity, config.TargetFrameworkVersion);
+                return (config.UnityInfo.Type, config.UnityInfo.Version);
             }
             else
             {
-                // TODO andrea: the correctness of these names need to be verified in projects that use each of the packages
-                // I didn't have any handy one.
-                var possibleFrameworks = new string[] { "Xamarin.Forms", "Mono.Android", "Xamarin.iOS", "Microsoft.Maui.Sdk" };
+                // the order in the array matters as all xamarin platforms share Xamarin.Forms.Platform
+                // but only Xamarin.Forms has only Xamarin.Forms.Platform
+                var possibleFrameworks = new string[]
+                {
+                    "Xamarin.Forms.Platform.UAP", "Microsoft.Maui",
+                    "Xamarin.Forms.Platform.Android", "Xamarin.Forms.Platform.iOS",
+                    "Xamarin.Forms.Platform.tvOS", "Xamarin.Mac",
+                    "Xamarin.Forms.Platform"
+                };
+
                 AssemblyNameReference frameworkUsedInConjunction = null;
                 foreach (var toSearch in possibleFrameworks)
                 {
-                    frameworkUsedInConjunction = module.FindReference(toSearch);
+                    frameworkUsedInConjunction = module.AssemblyReferences.Where(a => a.Name == toSearch).SingleOrDefault();
```

Comment:
Okay, I don't think this is going to work as intended. We have 3 frameworks we want to detect (in this order):

1. MAUI
2. Xamarin.Forms
3. Xamarin (native)
4. Catalyst
5. UWP

I think to achieve that we want something like:

```csharp
var possibleFrameworks = new[]
{
    "Microsoft.Maui", // MAUI
    "Xamarin.Forms.Core", // Xamarin.Forms
    "Xamarin.iOS", // Xamarin
    "Xamarin.tvOS", // Xamarin
    "Xamarin.Mac" // Xamarin
    "Mono.Android", // Xamarin
    "Microsoft.MacCatalyst", // Catalyst
    "Windows.Foundation.UniversalApiContract" // UWP
}
```